### PR TITLE
Value breakdowns: Update UI

### LINKED ISSUE #862: Drill in to label and field values could have a more visual change
I'ts easy to lose your place when you drill into a field or label to see the values. 
There are subtle changes on the screen but they are on your periphery and easy to miss. 
I've found myself lost more than a few times. 
https://github.com/user-attachments/assets/64f8afa5-5a17-497f-af9d-b96e426bc39f

### PR DESCRIPTION
**Pull request by gtk-grafana**
Fixes #862 

Adds "summary" time series above value breakdowns for visual differentiation of value breakdowns from label breakdowns.

image
image

Collapsed:
image

* ~Want to look into moving the field/label name selector into the panel header
image~
* ⌛ [Adding interactions to summary viz](https://github.com/grafana/explore-logs/issues/939)
* ✅  Sync text search 
* ✅ Allow collapse of summary panel
* ✅  Limit number of series rendered in summary panel
* ✅  [Sync series colors](https://github.com/grafana/explore-logs/pull/937)
* ✅  move search to above the summary panel
* ✅  remove 11.4 dependencies
* ✅  e2e coverage

### CODE REVIEW

**Comment by gtk-grafana** [comment]
With https://github.com/grafana/scenes/pull/992/ merged, I want to remove the Explore Logs specific series limit functionality as adding a series limit to the summary panel was getting unweildy due to the assumptions it was making about transformer/query runner relationships.

TL;DR: Want to upgrade scenes and implement https://github.com/grafana/scenes/pull/978 on main before taking this out of draft.

File: src/Components/Panels/PanelMenu.tsx [hunk 1]
```diff
@@ -166,20 +135,107 @@ export class PanelMenu extends SceneObjectBase<PanelMenuState> implements VizPan
...
+  items.push({
+    text: 'Visualization',
+    type: 'group',
+  });
+}
+
+function addCollapsableItem(items: PanelMenuItem[], menu: PanelMenu) {
+  const viz = sceneGraph.getAncestor(menu, VizPanel);
+  items.push({
+    text: viz.state.collapsed ? CollapsablePanelType.expanded : CollapsablePanelType.collapsed,
```
Comment (matyax) [inline src/Components/Panels/PanelMenu.tsx:152]: Seeing the usage here, I'm a bit confused by `CollapsablePanelType`. Is it storing copy? Is it a type? Does it need to be an enum?

Also, maybe more than a collapsable panel type, is a collapsable panel state, that can be expanded or collapsed?

Comment (matyax) [reply src/Components/Panels/PanelMenu.tsx:152]: Just saw line 155:

```typescript
    const newCollapsable State = viz.state.collapsed ? CollapsablePanel Type .expanded :
```

File: src/Components/ServiceScene/Breakdowns/ClearFiltersLayoutScene.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { SceneComponentProps, SceneObjectBase, SceneObjectState } from '@grafana/scenes';
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/ClearFiltersLayoutScene.tsx:1]: Maybe rename to `NoResultsScene` or `NoMatchingLabelsScene`? Also, does it need "Layout" as part of the name?

File: src/Components/ServiceScene/Breakdowns/ClearFiltersLayoutScene.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { GrotError } from '../../GrotError';
+import { Alert, Button } from '@grafana/ui';
+import React from 'react';
+import { emptyStateStyles } from './FieldsBreakdownScene';
+
+export interface ClearFiltersLayoutSceneState extends SceneObjectState {
+  clearCallback: () => void;
+}
+export class ClearFiltersLayoutScene extends SceneObjectBase<ClearFiltersLayoutSceneState> {
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/ClearFiltersLayoutScene.tsx:10]: We could also make the message configurable and reuse here https://github.com/grafana/grafana/blob/main/public/app/features/trails/Breakdown/ByFrameRepeater.tsx#L163

File: src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx [hunk 4]
```diff
@@ -336,34 +274,46 @@ export class FieldsBreakdownScene extends SceneObjectBase<FieldsBreakdownSceneSt
     navigateToValueBreakdown(ValueSlugs.field, value, serviceScene);
   };
 
-  public static Component = ({ model }: SceneComponentProps<FieldsBreakdownScene>) => {
-    const { body, loading, blockingMessage, search, sort } = model.useState();
+  public static ParentMenu = ({ model }: SceneComponentProps<FieldsBreakdownScene>) => {
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx:277]: What is a parent menu?

File: src/Components/ServiceScene/Breakdowns/LabelBreakdownScene.tsx [hunk 5]
```diff
@@ -273,34 +272,56 @@ export class LabelBreakdownScene extends SceneObjectBase<LabelBreakdownSceneStat
     navigateToValueBreakdown(ValueSlugs.label, value, serviceScene);
   };
 
-  public static Component = ({ model }: SceneComponentProps<LabelBreakdownScene>) => {
-    const { body, loading, blockingMessage, error, search, sort } = model.useState();
+  public static ParentMenu = ({ model }: SceneComponentProps<LabelBreakdownScene>) => {
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/LabelBreakdownScene.tsx:275]: For clarity, I would consider calling these `BreakdownMenu` and `ValuesMenu` or even `LabelsMenu` and `ValuesMenu`.

File: src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx [hunk 6]
```diff
@@ -31,13 +31,18 @@ import { DEFAULT_SORT_BY } from '../../../services/sorting';
...
+import { IndexScene } from '../../IndexScene/IndexScene';
+import { clearVariables, getVariablesThatCanBeCleared } from '../../../services/variableHelpers';
+import { ValueSummaryPanelScene } from './Panels/ValueSummary';
 
 type DisplayError = DataQueryError & { displayed: boolean };
 type DisplayErrors = Record<string, DisplayError>;
 
 export interface LabelValueBreakdownSceneState extends SceneObjectState {
-  body?: LayoutSwitcher & SceneObject;
+  body?: (LayoutSwitcher & SceneObject) | (ClearFiltersLayoutScene & SceneObject) | (EmptyLayoutScene & SceneObject);
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx:45]: Is the `& SceneObject` necessary?

File: src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx [hunk 7]
```diff
@@ -117,53 +118,73 @@ export class LabelValuesBreakdownScene extends SceneObjectBase<LabelValueBreakdo
         }
       }
     }
+  }
+
+  private setErrorStates(newState: SceneDataState) {
+    // If panels have errors
+    if (newState?.data?.errors && newState.data?.state !== LoadingState.Done) {
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx:125]: Should we consider when LoadingState is streaming?

File: src/Components/ServiceScene/Breakdowns/LayoutSwitcher.tsx [hunk 8]
```diff
@@ -15,15 +16,7 @@ export interface LayoutSwitcherState extends SceneObjectState {
...
-  public Selector({ model }: { model: LayoutSwitcher }) {
-    const { active, options } = model.useState();
-
-    return (
-      <Field>
-        <RadioButtonGroup options={options} value={active} onChange={model.onLayoutChange} />
-      </Field>
-    );
-  }
+  public static Selector = LayoutSwitcherComponent;
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/LayoutSwitcher.tsx:19]:
```suggestion
  public static Component = LayoutSwitcherComponent;
```
Does it make sense for it to be called `Selector`?

**Review by matyax — COMMENTED** [review]
Looks and works great! Left a few minor comments around.

File: src/Components/ServiceScene/Breakdowns/Panels/ValueSummary.tsx [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+            newState.collapsed ? CollapsablePanelType.collapsed : CollapsablePanelType.expanded
+          );
+        }
+      })
+    );
+  }
+}
+
+export function setValueSummaryHeight(vizPanelFlexLayout: SceneFlexLayout, collapsableState: CollapsablePanelType) {
+  const height = getValueSummaryHeight(collapsableState);
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/Panels/ValueSummary.tsx:84]: Just did a test out of curiosity and wonder why we need to manually set the height, otherwise it disappears.

File: src/Components/Panels/PanelMenu.tsx (hunk 1 above)
Comment (gtk-grafana) [reply src/Components/Panels/PanelMenu.tsx]: Yeah this is confusing, renamed to CollapsablePanelText: https://github.com/grafana/explore-logs/pull/936/commits/3159461a9996a44be3918afb33d547a49004b807

File: src/Components/ServiceScene/Breakdowns/ClearFiltersLayoutScene.tsx (hunk 3 above)
Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/ClearFiltersLayoutScene.tsx]: A bit confused, are you proposing we move this to core?

File: src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx (hunk 4 above)
Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx:277]: A poorly named menu :P

Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx]: See https://github.com/grafana/explore-logs/pull/936/commits/8feec08b0b26cb0773e299722172e14fe1ace406

File: src/Components/ServiceScene/Breakdowns/LabelBreakdownScene.tsx (hunk 5 above)
Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/LabelBreakdownScene.tsx]: Fixed: see https://github.com/grafana/explore-logs/pull/936/commits/8feec08b0b26cb0773e299722172e14fe1ace406

File: src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx (hunk 6 above)
Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx]: Yeah this shit always throws me too, this is my current understanding:

The discriminated union of intersecting types `{(A extends C) & C} | {(B extends C) & C}` can only be replaced by a straight up discriminated union `{A extends C} | {B extends C}` if all types in B and A are common

This is the error you get if you try to define
```
body?: (LayoutSwitcher & SceneObject) | (SceneReactObject & SceneObject);
```
instead of
```
body?: (LayoutSwitcher) | (SceneReactObject);
```

Error:
```
TS2322: Type LayoutSwitcher | SceneReactObject<{}> is not assignable to type SceneReactObject<{}> & LayoutSwitcher
Type LayoutSwitcher is not assignable to type SceneReactObject<{}> & LayoutSwitcher
Type LayoutSwitcher is not assignable to type SceneReactObject<{}>
Types of property Component are incompatible.
Type SceneComponent<LayoutSwitcher> is not assignable to type SceneComponent<SceneReactObject<{}>>
Property onLayoutChange is missing in type SceneReactObject<{}> but required in type LayoutSwitcher
```

TL;DR Since `LayoutSwitcher` has `onLayoutChange` method, and `SceneReactObject` does not, `LayoutSwitcher |  SceneReactObject` would only contain the common properties of both.

^ Should only work if LayoutSwitcher and SceneReactObject had a common interface?

File: src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx (hunk 7 above)
Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx:125]: Yes, but isn't that what's its doing by checking that the loading state is not done?
`newState.data?.state !== LoadingState.Done`
_____________________^ :P 

This hasn't been tested with sharding very well tho, I'll take a look before closing this out.

File: src/Components/ServiceScene/Breakdowns/LayoutSwitcher.tsx (hunk 8 above)
Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/LayoutSwitcher.tsx:19]: LayoutSwitcher already has a `Component`, would `LayoutSelector` be better?

File: src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx (hunk 7 above)
Comment (matyax) [reply src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx:125]: Yeah, that's my point. Some done things but with errors will have an Error state, or other times Error but state is still Streaming, which can cause unexpected behaviors.

File: src/Components/ServiceScene/Breakdowns/LayoutSwitcher.tsx (hunk 8 above)
Comment (matyax) [reply src/Components/ServiceScene/Breakdowns/LayoutSwitcher.tsx:19]: Nah, only valid if there was no Component property.

**Review by matyax — APPROVED** [review]
☄️

File: src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx (hunk 7 above)
Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/LabelValuesBreakdownScene.tsx:125]: Took me a bit to get it to fail, but yes, we are showing the error in each panel and as a toast when thrown, but only when sharding is done retrying.

image
